misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/EvaluationImportResolver.java

```diff
@@ -0,0 +1,181 @@
+        List<ModuleSymbol> matchingModuleSymbols = visibleModuleSymbols.stream().filter(moduleSymbol ->
+                moduleSymbol.id().orgName().equals(bImport.orgName())
+                        && moduleSymbol.id().moduleName().equals(bImport.moduleName()))
+                .collect(Collectors.toList());
+
+        if (matchingModuleSymbols.isEmpty()) {
+            capturedErrors.add(createEvaluationException(IMPORT_RESOLVING_ERROR, modulePrefix));
+            return;
+        } else if (matchingModuleSymbols.size() > 1) {
+            capturedErrors.add(createEvaluationException("Multiple modules found for import '" + modulePrefix + "'"));
+            return;
+        }
+
+        capturedImports.put(bImport.alias(), matchingModuleSymbols.get(0));
+    }
+
+    private void loadVisibleImports() {
+        if (!this.visibleImports.isEmpty()) {
+            return;
+        }
+        SyntaxTree documentTree = context.getDocument().syntaxTree();
+        if (!documentTree.containsModulePart()) {
+            return;
+        }
+
+        NodeList<ImportDeclarationNode> imports = ((ModulePartNode) documentTree.rootNode()).imports();
+        for (ImportDeclarationNode importNode : imports) {
+            Optional<String> orgName = resolveImportOrgName(importNode);
+            Optional<String> moduleName = importNode.moduleName().stream()
+                    .map(identifierToken -> encodeIdentifier(identifierToken.text().trim(), OTHER))
+                    .reduce((s1, s2) -> s1 + "." + s2);
+
+            if (orgName.isPresent() && moduleName.isPresent()) {
+                String importAlias = resolveImportAlias(importNode, moduleName.get());
+                BImport bImport = new BImport(orgName.get(), moduleName.get(), importAlias);
+                visibleImports.put(importAlias, bImport);
+            }
+        }
+    }
+
+    private void loadVisibleModuleSymbols() {
+        if (!this.visibleModuleSymbols.isEmpty()) {
+            return;
+        }
+        visibleModuleSymbols.addAll(context.getSemanticInfo().visibleSymbols(context.getDocument(), LinePosition
+                .from(context.getLineNumber(), 0))
+                .stream()
+                .filter(symbol -> symbol.kind() == SymbolKind.MODULE)
+                .map(symbol -> ((ModuleSymbol) symbol))
+                .collect(Collectors.toList()));
+    }
+
+    private Optional<String> resolveImportOrgName(ImportDeclarationNode importNode) {
+        if (importNode.orgName().isPresent()) {
+            return Optional.of(importNode.orgName().get().orgName().text().trim());
+        } else {
+            return context.getPackageOrg();
+        }
+    }
+
+    private String resolveImportAlias(ImportDeclarationNode importNode, String moduleName) {
+        if (importNode.prefix().isPresent()) {
+            return importNode.prefix().get().prefix().text().trim();
+        } else {
+            String[] moduleNameParts = moduleName.split("\\.");
```

Comment:
Refer this [comment](https://github.com/ballerina-platform/ballerina-lang/pull/32621/files#r704925279)